Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -1769,6 +1783,8 @@ theorem exists_nat_eq_of_le_nat {c : Cardinal} {n : ℕ} (h : c ≤ n) : ∃ m,
   ⟨toNat c, natCast_le.1 (he.trans_le h), he.symm⟩
 #align cardinal.exists_nat_eq_of_le_nat Cardinal.exists_nat_eq_of_le_nat
 
+lemma mk_toNat_of_isEmpty [IsEmpty α] : toNat #α = 0 := by simp
+
```

Comment:
It's not clear to me that this lemma is useful, especially since `simp` proves it.

Reply:
I'm not so sure, because it relies on `IsEmpty α → Fintype α` being an instance, and this is arguably quite bad. Certainly no one in their right mind would find the proof by guessing that they can use that `α` is a fintype, and they would instead expect the analogue of `mk_toNat_of_infinite` to exist.

Reply:
Surely `simp` proceeds by showing ` [IsEmpty α] : #α = 0`? If that lemma doesn't exist, then *that* should be the simp lemma.

Reply:
Oh you're right. `mk_eq_zero`.